lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -175,8 +175,20 @@ public function fetchAll($fetchMode = null, $columnIndex = 0)
             return $rows;
         }
 
-        foreach ($rows as $num => $row) {
-            $rows[$num] = $this->fixRow($row, $iterateRow, $fixCase);
+        if ($fetchMode === PDO::FETCH_COLUMN) {
+            if ($iterateRow) {
+                foreach ($rows as $num => $value) {
+                    if (($this->portability & Connection::PORTABILITY_EMPTY_TO_NULL) && $value === '') {
+                        $rows[$num] = null;
+                    } elseif (($this->portability & Connection::PORTABILITY_RTRIM) && is_string($value)) {
```

Comment:
The first bitwise expression does not need re-computing within the loop, and should be moved to a variable